### ISSUE TITLE
Regression: new false positive in `no-done-callback`

### ISSUE THREAD

**Issue by wjhsf** [vitest-dev/eslint-plugin-vitest#802]
PR #800 (released in [v1.3.24](https://github.com/vitest-dev/eslint-plugin-vitest/releases/tag/v1.3.24)) fixed a false positive in `no-standalone-expect`, but it introduced a new one in `no-done-callback`.

```js
import { test, expect } from 'vitest';
test.for([1, 2, 3])('%s is a number', (n) => {
    expect(n).toBeTypeOf('number');
});
```

The above code passes validation with v1.3.23, but fails with v1.3.24.
